core/app.cc

```diff
@@ -319,22 +326,27 @@ void application::main()
     elf::get_program()->init_library(_args.size(), _argv.get());
     sched::thread::current()->set_name(_command);
 
-    if (_main) {
-        run_main();
+    if (_lib->is_statically_linked()) {
+        elf_entry_point(_lib->entry_point(), _args.size(), _argv.get(), _argv_size);
```

Comment:
Continuing my previous comment, if the code in the contructor had set _entry_point like it did for the PIE case, you could have used _entry_point here instead of _lib->entry_point().

Reply:
Addressed in the latest version of this PR.